akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -383,29 +382,20 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
     byteBuffer.clear()
 
     // Write fixed length parts
-    byteBuffer.put(header.version)
-    byteBuffer.put(header.flags)
-    // 1 empty byte slot // TODO avoid having these empty slots
-    // 1 empty byte slot
-    byteBuffer.position(UidOffset) // skips the above 2 empty slots
-    byteBuffer.putLong(header.uid)
-    byteBuffer.putInt(header.serializer)
+    byteBuffer.put(VersionOffset, header.version)
```

Comment:
The reason for explicit offsets was because alignment IIRC. I.e if you write a byte it will move the offset with only one byte, but I wanted fields on 4-8 byte boundaries and I would prefer them on these boundaries if possible.